interface-impl-amf/src/test/java/org/mule/amf/impl/model/ParameterImplTest.java

```diff
@@ -91,6 +91,7 @@ public void nonNullableInteger() {
       assertEquals("expected type: Number, found: Null", nonNullableInteger.message(null));
     }
     assertTrue(nonNullableInteger.validate("123"));
+    assertFalse(nonNullableInteger.validate("123\n123"));
```

Comment:
I would add a test for encoded and special characters apart from the non-nullable use case.